src/components/DeepImageInput/index.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { useCallback, useState } from 'react';
+import { _cs } from '@togglecorp/fujs';
+import { useLazyRequest } from '#utils/request';
+import { FileInput, FileInputProps, ImagePreview } from '@the-deep/deep-ui';
+import _ts from '#ts';
+
+interface Option {
+    id: number;
+    title: string;
+    file: string;
+    mimeType?: string;
+    metadata?: unknown;
+}
+
+interface Props<T extends string> extends Omit<FileInputProps<T>, 'overrideStatus' | 'status' | 'value' | 'onChange' | 'multiple' | 'className'> {
+    className?: string;
+    previewClassName?: string;
+    fileInputClassName?: string;
+    value?: number;
+    onChange: (value: number | undefined, name: T) => void;
+
+    option?: Option;
+    setOption: (value: Option) => void;
+    isPublic?: boolean;
+    showPreview?: boolean;
```

Comment:
The prop name should not be named as a function name.
Let' rename this to something like `previewShown` or `previewEnabled` or `previewVisible`